src/txn/txn_rollback_to_stable.c

```diff
@@ -30,6 +30,12 @@ __rollback_abort_newer_update(WT_SESSION_IMPL *session, WT_UPDATE *first_upd,
         if (upd->txnid == WT_TXN_ABORTED || upd->start_ts == WT_TS_NONE) {
             if (upd == first_upd)
                 first_upd = upd->next;
+
```

Comment:
could you reorganise the code?  something like
if (upd->txnid == WT_TXN_ABORTED)
      continue;
if (upd->start_ts == WT_TS_NONE)
    ...

Reply:
Fixed. I removed the case of checking the non-timestamp to move the first valid update. I don't think it is required.

Reply:
Further, an update is made to the function to stop processing the updates when a valid update is found. Still, it is possible to remove the `first_upd` variable, but currently, it is used in some diagnostic cases, so kept it as it is.